dvc/remote/local/__init__.py

```diff
@@ -573,3 +573,52 @@ def _get_unpacked_dir_names(self, checksums):
             if self.is_dir_checksum(c):
                 unpacked.add(c + self.UNPACKED_DIR_SUFFIX)
         return unpacked
+
+    def _needs_checkout(self, path_info, checksum, force):
+        # NOTE: In case if path_info is already cached, and cache type is
+        # 'copy' we would like to avoid relinking.
+        # `symlink/hardlink/reflink` is cheap, but if we need `copy` links,
+        # then to save on copying we should only relink if we already have
+        # `hardlink` or a `symlink`, we don't care about reflinks, because
+        # they are indistinguishable from copy anyway.
+
+        if (
+            not force
```

Comment:
Not quite sure that `force` is applicable here. In _checkout_file, force is used to silently remove a file if it is not in the cache, and here we are reusing it for forcing relinking.

Reply:
Yeah, I guess there is no point in keeping it here